symengine/dense_matrix.cpp

```diff
@@ -364,76 +364,96 @@ void mul_dense_scalar(const DenseMatrix &A, const RCP<const Basic> &k,
 }
 
 // ---------------------------- Joining Operations ---------------------------//
-void row_join(const DenseMatrix &A, const DenseMatrix &B, DenseMatrix &C)
+void DenseMatrix::row_join(const DenseMatrix &B)
 {
-    SYMENGINE_ASSERT(A.row_ == B.row_
-                     and (A.row_ == C.row_ and C.col_ == A.col_ + B.col_));
+    this->col_insert(B, col_);
+}
 
-    unsigned row = A.row_, col = A.col_;
+void DenseMatrix::col_join(const DenseMatrix &B)
+{
```

Comment:
Since it is already checked in `row_insert` and `col_insert`, I think it will be redundant. Same for above.